In [27]:
import pandas as pd
import numpy as np

## Do all three analyses of UCI data return consistent features?

In [3]:
feature_results_uci_1 = pd.read_pickle("feature_results_uci_1.pickle")
feature_results_uci_2 = pd.read_pickle("feature_results_uci_2.pickle")
feature_results_uci_3 = pd.read_pickle("feature_results_uci_3.pickle")

In [11]:
uci_1_related_features = feature_results_uci_1.sort_values('R2', ascending = False).head(20).index
uci_2_related_features = feature_results_uci_2.sort_values('R2', ascending = False).head(20).index
uci_3_related_features = feature_results_uci_3.sort_values('R2', ascending = False).head(20).index

In [28]:
uci_1_related_features = np.array(uci_1_related_features.sort_values())
uci_2_related_features = np.array(uci_2_related_features.sort_values())
uci_3_related_features = np.array(uci_3_related_features.sort_values())

In [30]:
uci_1_related_features == uci_2_related_features

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True], dtype=bool)

In [31]:
uci_1_related_features == uci_3_related_features

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True], dtype=bool)

In [32]:
uci_3_related_features == uci_2_related_features

array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True], dtype=bool)

In [33]:
uci_1_related_features

array([ 28,  48,  64, 105, 128, 153, 241, 281, 318, 336, 338, 378, 433,
       442, 451, 453, 455, 472, 475, 493])

### UCI Conclusion
Each sample of UCI data suggests that the same 20 features are related, giving us high confidence that the following features are predictors of the target: 

`[28,  48,  64, 105, 128, 153, 241, 281, 318, 336, 338, 378, 433, 442, 451, 453, 455, 472, 475, 493]`